src/normalizer.ts

```diff
@@ -99,62 +98,228 @@ function normalizeV0Schema(parsed: parser.V0Schema): XtpSchema {
   })
 
   return {
-    version,
-    exports,
-    imports,
-    schemas,
+    schema: {
+      version: 'v0',
+      exports,
+      imports,
+      schemas,
+    },
+    errors: []
   }
 }
 
+interface CycleDetectionContext {
+  visited: Set<string>;
+  stack: Set<string>;
+  path: string[];
+  errors: ValidationError[];
+}
+
+function detectSchemaRefCycles(
```

Comment:
Something about this function feels too big. I wrote some code to workaround cycles by storing cycles then processing them later. so part of that was detecting cycles and storing them. and it was only a few lines of code. Perhaps i was missing something though. Either way i think this is good because it's separate from the main parse and normalize code so we can easily remove it later.